main/views.py

```diff
@@ -213,6 +221,38 @@ def form_valid(self, form):
             return redirect('signup')
 
 
+def space_approval(request, key, action):
+
+    if action != 'approve' and action != 'reject':
+        # this shouldn't happen - just redirect to home
+        return redirect('/')
+
+    # lookup user info based on key
+    users = User.objects.filter(space_request_key=key)
+
+    # should only get one object back
+    if users.count() == 1:
```

Comment:
`User.objects.get(space_request_key=key)` is a more idiomatic way of doing this - it will throw an exception if the object doesn't exist ([docs](https://docs.djangoproject.com/en/1.11/topics/db/queries/#retrieving-a-single-object-with-get))